lib/storage/src/rbac/mod.rs

```diff
@@ -1,9 +1,209 @@
-pub mod access;
-pub mod access_level;
-pub mod access_token;
-#[allow(dead_code)]
-pub mod collection_access;
-#[allow(dead_code)]
-pub mod error;
-
 mod ops_checks;
+
+use std::borrow::Cow;
+use std::collections::HashMap;
+
+use segment::json_path::JsonPath;
+use segment::types::ValueVariants;
+use serde::{Deserialize, Serialize};
+
+use crate::content_manager::errors::StorageError;
+
+/// A structure that defines access rights.
+#[derive(Serialize, Deserialize, PartialEq, Clone, Debug)]
+#[serde(untagged)]
+pub enum Access {
+    /// Cluster-wide access.
+    Cluster(ClusterAccessMode),
+    /// Access to specific collections.
+    Collection(AccessCollection),
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Clone, Debug)]
+pub struct AccessCollection(pub Vec<AccessCollectionRule>);
+
+#[derive(Serialize, Deserialize, PartialEq, Clone, Debug)]
+pub struct AccessCollectionRule {
+    /// Collection names that are allowed to be accessed
+    pub collections: Vec<String>,
+
+    pub access: CollectionAccessMode,
+
+    /// Payload constraints.
+    /// An object where each key is a JSON path, and each value is JSON value.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub payload: Option<PayloadConstraint>,
+}
+
+#[derive(Serialize, Deserialize, Eq, PartialEq, Ord, PartialOrd, Copy, Clone, Debug)]
+pub enum ClusterAccessMode {
```

Comment:
Why is this name specific to the Cluster concept?
```suggestion
pub enum GlobalAccessMode {
```